crm_lead_propagate_lost/wizards/crm_lead_lost.py

```diff
@@ -0,0 +1,17 @@
+###############################################################################
+# For copyright and license notices, see __manifest__.py file in root directory
+###############################################################################
+from odoo import api, models
+
+
+class CrmLeadLost(models.TransientModel):
+    _inherit = 'crm.lead.lost'
+
+    @api.multi
+    def action_lost_reason_apply(self):
+        leads = self.env['crm.lead'].browse(self.env.context.get('active_ids'))
+        for lead in leads:
+            for sale_order in lead.order_ids.filtered(
+                    lambda s: s.state != 'cancel'):
+                sale_order.action_cancel()
```

Comment:
Yo yo simplificaría así, hace lo mismo, sin el filterred claro, pero me daría igual cancelar algo ya cancelado, no hará nada:
```suggestion
            lead.order_ids.action_cancel()
```